Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -120,10 +123,11 @@ public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder)
                     $value = $this->getFilterValueFromUrl($value);
                 }
 
-                $valueParameter = QueryUtils::generateParameterName($field);
+                $valueParameter = QueryUtils::generateParameterName($this->isPropertyNested($property) ? $alias.'_'.$field : $field);
```

Comment:
Imagine the following story (introducing _Dummy.name_ & _RelatedDummy.name_ fields):
> I wanna filter Dummy by name, and RelatedDummy by name

Using your solution, there will be a conflict on parameter `name_123456abcdefg` used for both criteria. Using mine, there will be `name_123456abcdefg` and `relatedDummy_123456abcdefg_name_123456abcdefg` parameters, so no conflict.